src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,503 @@
+            self.scm_namespace_set.add(blockdev)
+
+        self.log.info(
+            "scm_socket_to_namespaces = %s", self.scm_socket_to_namespaces)
+        self.log.info("scm_namespace_set = %s", self.scm_namespace_set)
+
+        # Call dmg network scan --provider=all --json.
+        network_out = dmg.network_scan(provider="all")
+
+        # Check the status.
+        status = network_out["status"]
+        if status != 0:
+            self.log.error(network_out["error"])
+            self.fail("dmg network scan failed!")
+
+        # Get the hash and the Interfaces list.
+        temp_dict = network_out["response"]["HostFabrics"]
+        struct_hash = list(temp_dict.keys())[0]
+        interfaces = temp_dict[struct_hash]["HostFabric"]["Interfaces"]
+
+        # Fill in the dictionary and the set for interface.
+        for interface in interfaces:
+            provider = interface["Provider"]
+            device = interface["Device"]
+            self.interface_to_providers[device].append(provider)
+            self.interface_set.add(device)
+
+        self.log.info(
+            "interface_to_providers = %s", self.interface_to_providers)
+        self.log.info("interface_set = %s", self.interface_set)
+
+    def check_errors(self, errors):
+        """Check if there's any error in given list. If so, fail the test.
+
+        Args:
+            errors (list): List of errors.
+        """
+        if errors:
+            self.fail("\n----- Errors detected! -----\n{}".format(
+                "\n".join(errors)))
+
+    def verify_access_point(self, host_port_input, failure_expected):
+        """Run with given AP and verify the AP in the output.
+
+        Args:
+            host_port_input (str): Host:Port or just Host. Supports multiple APs
+                that are separated by comma.
+            failure_expected (bool): Failure is expected in this run.
+
+        Returns:
+            list: List or errors.
+        """
+        errors = []
+        check = {}
+
+        check["expected"] = host_port_input.split(",")
+        if ":" not in host_port_input:
+            # dmg automatically sets 10001 if it's not given in the input.
+            check["expected"] = [
+                "{}:10001".format(host) for host in check["expected"]]
+
+        try:
+            generated_yaml = self.get_dmg_command().config_generate(
+                access_points=host_port_input)
+            check["actual"] = generated_yaml["access_points"]
+            if failure_expected:
+                errors.append("dmg expected to fail, but it worked!")
+            elif sorted(check["expected"]) != sorted(check["actual"]):
+                errors.append(
+                    "Unexpected access point: {} != {}".format(
+                        check["expected"], check["actual"]))
+        except CommandFailure as err:
+            if not failure_expected:
+                errors.append("Unexpected failure! {}".format(err))
```

Comment:
It might be useful to verify the failed command fails for the correct reason.   This would also require updating https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/dmg_utils.py#L895 to return the CmdResult object from `run()`:
```suggestion
        my_dmg = DmgCommand(self.bin)
        my_dmg.exit_status_exception = False
        try:
            result = my_dmg.config_generate(access_points=host_port_input)
        except CommandFailure as err:
            errors.append("Unexpected failure! {}".format(err))
            
        if result.exit_status == 0 and failure_expected is None:
            try:
                yaml_data = yaml.safe_load(result.stdout)
                check["actual"] = yaml_data["access_points"]
                if sorted(check["expected"]) != sorted(check["actual"]):
                    errors.append(
                        "Unexpected access point: {} != {}".format(
                            check["expected"], check["actual"]))
            except yaml.YAMLError as error:
                errors.append(
                    "Error loading dmg generated config: {}".format(error))
        elif result.exit_status == 0 and failure_expected is not None:
            errors.append(
                    "dmg command passed when expected to fail: {}".format(result))
        elif result.exit_status != 0 and failure_expected is not None:
            if failure_expected not in result.stderr_text:
                errors.append("Missing expected error message in failed dmg command: {}".format(result))
        else:
            errors.append(
                    "dmg command failed when expected to pass: {}".format(result))
```